TLT-2664: course selection text and link updates

elliottyates opened the pull request:
(Opening this for feedback -- text changes still to come)

Updates old shopping JS to new course selection process.
- removes manage shopping left-hand nav item
- dev, prod terms, role IDs, etc are controlled with a single variable, for easier testing
- text changes (shopping --> course selection; shopper --> prospective enrollee; etc)
- add and remove links updated to use my.harvard instead of old shopping tool
- course URLs using sis_course_id are supported

elliottyates committed:
tlt-2664: course selection text and link updates

elliottyates committed:
tlt-2664: course selection - removes manage shopping nav item

elliottyates committed:
tlt-2664: course selection text updates for prospective enrollees

elliottyates committed:
tlt-2664: course selection updates
- prod fall term updated, dev term noted in comment
- role check switched to role_ids where appropriate, fixes issue where role name changed from shopper to prospective enrollee
- shopping terminology changed to course selection in code
- minor refactoring to move variables into more local scopes

elliottyates committed:
tlt-2664: course selection improvements
- add course button uses icommons_tools.course_selection app to redirect to appropriate my.harvard course
- course selection functions and variables are moved outside of global namespace (also supports easier unit testing in future)
- course URLs using sis_course_id are supported
- single flag to indicate whether to use test or prod values, to speed up testing

eparker71 commented on js/shop.js:
```diff
@@ -1,299 +1,312 @@
+var huCourseSelection = (function (){
+  var currentEnvironment='prod';
+  var devProspectiveEnrolleeRole = '11';
```
super nit pick, I might call this devProspectiveEnrolleeRoleId to indicate that it's an Id vs. a role name. not critical at all.

eparker71 commented on js/shop.js:
```diff
@@ -1,299 +1,312 @@
+   */
+  var course_selection_help_doc_url = 'https://wiki.harvard.edu/confluence/pages/viewpage.action?pageId=168134774';
+  var data_tooltip = 'More info about access during course selection period';
+  var tooltip_link = '<a data-tooltip title="' + data_tooltip + '" target="_blank" href="' +
```
Just a stylistic comment, I'm used to seeing things defined before they are used. so seeing this below where it's used is a bit confusing. Same it true for all the vars used this way add_course_url, remove_course_url etc...

eparker71 commented:
👍 looks good, I had a couple of comments but they are really stylistic in nature, not critical.

elliottyates commented:
Note: if we merge this into `master`, it'll mean any deploys will include (and therefore 'turn on') course selection banners for the prod terms in the file. We may want to rename it to `js.hold` before the merge, if it's going to be well ahead of the course selection period start date for this school.

elliottyates committed:
tlt-2664: course selection - turn on styling for course selection notices

cmurtaugh commented:
I'd say go ahead and merge it to master, and we can tag it with v1.5.  Since deploying involves a manual file upload into Canvas, I don't think there's much risk of causing problems.

elliottyates committed:
tlt-2664, tlt-2563: course selection text, styling, and functionality updates
- fix to enrolled student check
- text updates as of 8/10/2016
- styling updates as per tlt-2563

elliottyates committed:
tlt-2664: course selection - more clarity in variable names

eparker71 commented:
👍
